torchgeo/datasets/zuericrop.py

```diff
@@ -0,0 +1,234 @@
+
+        tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+        # Convert from TxHxWxC to TxCxHxW
+        tensor = tensor.permute((0, 3, 1, 2))
+        return tensor
+
+    def _load_target(self, index: int) -> Tuple[Tensor, Tensor, Tensor]:
+        """Load the target mask for a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target mask and label for each mask
+        """
+        import h5py
+
+        with h5py.File(self.filepath, "r") as f:
+            mask_array = f["gt"][index, ...]
+            instance_array = f["gt_instance"][index, ...]
+
+        mask_tensor = torch.from_numpy(mask_array)  # type: ignore[attr-defined]
+        instance_tensor = torch.from_numpy(instance_array)  # type: ignore[attr-defined]
+
+        # Convert from HxWxC to CxHxW
+        mask_tensor = mask_tensor.permute((2, 0, 1))
+        instance_tensor = instance_tensor.permute((2, 0, 1))
+
+        # Convert instance mask of N instances to N binary instance masks
+        instance_ids = torch.unique(instance_tensor)  # type: ignore[attr-defined]
+        # Exclude a mask for unknown/background
+        instance_ids = instance_ids[instance_ids != 0]
+        instance_ids = instance_ids[:, None, None]
+        masks: Tensor = instance_tensor == instance_ids
+
+        # Parse labels for each instance
+        labels_list = []
+        for mask in masks:
+            label = mask_tensor[mask[None, :, :]]
+            label = torch.unique(label)[0]  # type: ignore[attr-defined]
+            labels_list.append(label)
+
+        # Get bounding boxes for each instance
+        boxes_list = []
+        for mask in masks:
+            pos = torch.where(mask)  # type: ignore[attr-defined]
+            xmin = torch.min(pos[1])  # type: ignore[attr-defined]
+            xmax = torch.max(pos[1])  # type: ignore[attr-defined]
+            ymin = torch.min(pos[0])  # type: ignore[attr-defined]
+            ymax = torch.max(pos[0])  # type: ignore[attr-defined]
+            boxes_list.append([xmin, ymin, xmax, ymax])
+
+        masks = masks.to(torch.uint8)  # type: ignore[attr-defined]
+        boxes = torch.tensor(boxes_list).to(torch.float)  # type: ignore[attr-defined]
+        labels = torch.tensor(labels_list).to(torch.long)  # type: ignore[attr-defined]
+
+        return masks, boxes, labels
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the files already exist
+        exists = []
+        for filename in self.filenames:
+            filepath = os.path.join(self.root, filename)
+            exists.append(os.path.exists(filepath))
+
+        for e, filename in zip(exists, self.filenames):
+            filepath = os.path.join(self.root, filename)
+            print(e, filepath)
```

Comment:
It looks like this is debugging code that can be removed?

Reply:
Removed